### PR TITLE
core/mount: stop generating mount units for cred mounts

### PR DESCRIPTION
**Pull request by YHNdnzj**
Material for v257.

### CODE REVIEW

**Review by bluca — APPROVED** [review]
Having hundreds of mount units is a pain, so this sounds like the best solution to me.

**Comment by yuwata** [comment]
I have not reviewed this in detail, but the basic concept looks good to me.

File: src/core/mount.c [hunk 1]
```diff
@@ -1840,9 +1841,8 @@ static int mount_setup_unit(
         assert(options);
         assert(fstype);
 
-        /* Ignore API mount points. They should never be referenced in
-         * dependencies ever. */
-        if (mount_point_is_api(where) || mount_point_ignore(where))
+        /* Ignore API mount points. They should never be referenced in dependencies ever. */
```
Comment (keszybz) [inline src/core/mount.c]: "Ignore API mount points and \<something\>." Right now the comment doesn't match the code.

**Review by keszybz — APPROVED** [review]
Me likes very much.

**Pull request merged**